smart-contracts/assembly/contracts/website-deployer/websiteStorer.ts

```diff
@@ -0,0 +1,74 @@
+/**
+ * Website storer implementation by MassaLabs.
+ *
+ * This smartcontract is used to upload and manage website chunks.
+ *
+ */
+
+import {
+  Storage,
+  Context,
+  generateEvent,
+  callerHasWriteAccess,
+} from '@massalabs/massa-as-sdk';
+import {
+  Args,
+  bytesToI32,
+  i32ToBytes,
+  stringToBytes,
+} from '@massalabs/as-types';
+import { onlyOwner } from '../utils';
+import { _setOwner } from '../utils/ownership-internal';
+
+export const NB_CHUNKS_KEY = stringToBytes('NB_CHUNKS');
+
+export function constructor(_: StaticArray<u8>): void {
+  assert(callerHasWriteAccess());
```

Comment:
I think you should use [isDeployingContract](https://as-sdk.docs.massa.net/functions/Context.isDeployingContract.html) instead because the test is clearer.